index.js

```diff
@@ -57,14 +58,38 @@ Lexer.prototype = {
     if (!value) this.error('ASSERT_FAILED', message);
   },
 
-  assertExpression: function (exp) {
+  assertExpression: function (exp, noThrow) {
     //this verifies that a JavaScript expression is valid
+    var src = 'return (' + exp + ')';
     try {
-      Function('', 'return (' + exp + ')');
+      acorn.parse(src, {
+        ecmaVersion: 6,
+        allowReturnOutsideFunction: true
+      });
```

Comment:
We could go one step further by asserting that acorn.parse returns exactly one statement of type return statement.  That prevents things like `true)console.log('oops')(` from being seen as a valid expression.